---
layout: post
title:  "First group meeting"
date:   2017-11-27 18:00:00
categories: main
---
### Project planning
At the first group meeting, all group members were present and the project background and aim was discussed to make sure everyone has understood what the project is about. The main goal of this meeting was to organise the work for the upcoming days before the presentation on Seminar 1 (30th of November). It was decided that all project work (except for the individual diaries) is documented using Google Suite, i.e. Google Docs, Google sheets and Google Presentation. 

It was decided that an initial project plan, the Seminar 1 presentation, and basic read-up on assemblers and assembly approaches in general, should be done at the latest the evening of the 29th of November. An update on the presentation and project plan will be available in the next post. 

### Initial ideas for the project
As the main task is to assemble the genome of *P. andina*, we discussed different approaches and naturally what assembler we would use. A list of all assemblers available on UppMax was created and each assembler will be evaluated for the next coming 2 days in order to see which one fits the project best, and perhaps most importantly the data. 

As the genome of the eukaryote *P. infestans* is around 240 Megabases (the largest genome of all known Oomycetes) we assume the genome of *P. andina* will be of similar size (being its closest relative). We discussed three main criteria that our assembler should fulfill: 

- Be able to handle Eukaryotic genomes and all that comes with it, e.g. repetitive regions
- Handle short reads 
- Handle medium-sized genomes (the genome is not huge, but not very small either). 

Since we will have to annotate genes, BLAST is the first tool that comes to mind. We decided that we would already now look into how BLAST can be used for this task. 